src/waypoint.jsx

```diff
@@ -34,11 +34,24 @@ function getCurrentPosition(bounds) {
     return POSITIONS.invisible;
   }
 
+  // top is within the screen
   if (bounds.viewportTop <= bounds.waypointTop &&
       bounds.waypointTop <= bounds.viewportBottom) {
     return POSITIONS.inside;
   }
 
+  // bottom is within the screen
+  if (bounds.viewportTop <= bounds.waypointBottom &&
+      bounds.waypointBottom <= bounds.viewportBottom) {
+    return POSITIONS.inside;
+  }
+
+  // top is above the screen and bottom is below the screen
+  if (bounds.waypointTop <= bounds.viewportTop &&
+      bounds.viewportBottom <= bounds.waypointBottom) {
```

Comment:
Doesn't matter too much, but we could get away with these just be `<`. The `==` case should be handled by the above two conditionals.